packages/core/src/lib/implementation/execute-plugin.ts

```diff
@@ -148,21 +147,22 @@ export async function executePlugins(
 
   logMultipleResults(results, 'Plugins', undefined, errorsCallback);
 
-  // TODO: add groupBy method for promiseSettledResult by status, #287
-
-  const failedResults = results.filter(isPromiseRejectedResult);
-  if (failedResults.length) {
-    const errorMessages = failedResults
+  const { fulfilled, rejected } = groupByStatus(results) as {
+    fulfilled: PromiseFulfilledResult<PluginReport>[];
+    rejected: PromiseRejectedResult[];
+  };
+  if (rejected.length) {
+    const errorMessages = rejected
       .map(({ reason }: PromiseRejectedResult) => reason)
       .join(', ');
```

Comment:
Unnecessary parameter type annotation:

```suggestion
    const errorMessages = rejected.map(({ reason }) => reason).join(', ');
```